### PR TITLE
Increase log levels to diagnose MockedTypeSupport failures

### PR DESCRIPTION
**Pull request by jrw972**

### CODE REVIEW

**Review by mitza-oci — APPROVED** [review]

File: tests/DCPS/AddressLookup/main.cpp [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+  }
+
+  for (size_t i = 0; i < addr_count; ++i) {
+    ACE_DEBUG((LM_DEBUG, "address_info: Considering interface %d\n", i));
+    //ACE_TCHAR buffer[256];
+    //if (addr_array[i].addr_to_string(buffer, sizeof buffer) != 0) {
+    char buffer[256] = {'\0'};
+    if (addr_array[i].get_host_addr(buffer, sizeof(buffer)) == 0) {
+      ACE_ERROR((LM_ERROR, "ERROR: address_info: Failed to convert address to string\n"));
+    }
```
Comment (mitza-oci) [inline tests/DCPS/AddressLookup/main.cpp:132]: This looks like it does the same thing twice

Comment (sonndinh) [reply tests/DCPS/AddressLookup/main.cpp:132]: The first loop just gives overall information about the interfaces while the second prints out more details regarding DNS resolution for each interface. I thought it's helpful to have a summary of the interfaces first, but we can remove the first loop if it's redundant.

File: tests/DCPS/AddressLookup/main.cpp [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+#include <ace/OS_NS_string.h>
+#include <ace/OS_NS_unistd.h>
+
+#include <string>
+#include <cstring>
+
+
+void print_addr(const ACE_INET_Addr& addr, const char* str) {
+  //ACE_TCHAR buffer[256];
+  //if (addr.addr_to_string(buffer, sizeof buffer) != 0) {
```
Comment (sonndinh) [inline tests/DCPS/AddressLookup/main.cpp:14]:
```suggestion
```

File: tests/DCPS/AddressLookup/main.cpp [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+    } addr;
+    std::memset(&addr, 0, sizeof(addr));
+
+#ifdef ACE_HAS_IPV6
+    ACE_DEBUG((LM_DEBUG, "ip46.in6_ size is %d\n", sizeof(addr.in6_)));
+#endif
+    ACE_DEBUG((LM_DEBUG, "addr size is %d, curr addrinfo size is %d, curr->ai_addrlen is %d, curr->ai_addr size is %d, curr family is %d\n",
+               sizeof(addr), sizeof(*curr), curr->ai_addrlen, sizeof(*(curr->ai_addr)), curr->ai_family));
+    std::memcpy(&addr, curr->ai_addr, curr->ai_addrlen);
+    //std::memcpy(&addr, curr->ai_addr, sizeof addr);
```
Comment (sonndinh) [inline tests/DCPS/AddressLookup/main.cpp:84]:
```suggestion
```

File: tests/DCPS/AddressLookup/main.cpp [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+      }
+      else {
+          ACE_DEBUG((LM_DEBUG, "DEBUG: address_info: Found IP interface %C\n", buffer));
+      }
+  }
+
+  for (size_t i = 0; i < addr_count; ++i) {
+    ACE_DEBUG((LM_DEBUG, "address_info: Considering interface %d\n", i));
+    //ACE_TCHAR buffer[256];
+    //if (addr_array[i].addr_to_string(buffer, sizeof buffer) != 0) {
```
Comment (sonndinh) [inline tests/DCPS/AddressLookup/main.cpp:128]:
```suggestion
```

File: tests/DCPS/AddressLookup/main.cpp [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+    return;
+  }
+
+  ACE_DEBUG((LM_DEBUG, "address_info: There are %d interfaces\n", addr_count));
+  for (size_t i = 0; i < addr_count; ++i) {
+      char buffer[256] = { '\0' };
+      if (addr_array[i].get_host_addr(buffer, sizeof(buffer)) == 0) {
+          ACE_ERROR((LM_ERROR, "ERROR: address_info: Failed to convert address to string\n"));
+      }
+      else {
```
Comment (sonndinh) [inline tests/DCPS/AddressLookup/main.cpp:120]:
```suggestion
      } else {
```

File: tests/DCPS/AddressLookup/main.cpp [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+      ACE_ERROR((LM_ERROR, "ERROR: address_info: Failed to convert address to string\n"));
+    }
+
+    // Find the hostname of the interface
+    char hostname[MAXHOSTNAMELEN+1] = {'\0'};
+    if (ACE::get_fqdn(addr_array[i], hostname, MAXHOSTNAMELEN+1) == 0) {
+      ACE_DEBUG((LM_DEBUG, "DEBUG: address_info: IP address %C maps to hostname %C\n", buffer, hostname));
+    } else {
+      ACE_ERROR((LM_ERROR, "ERROR: address_info: Failed to get FQDN\n"));
+    }
```
Comment (sonndinh) [inline tests/DCPS/AddressLookup/main.cpp:140]:
```suggestion
    } else {
      ACE_ERROR((LM_ERROR, "ERROR: address_info: Failed to get FQDN\n"));
      continue;
    }
```
Skip this interface if it fails to get FQDN.

File: tests/DCPS/AddressLookup/main.cpp [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+    temp.set_port_number(port_number, 1 /*encode*/);
+
+    print_addr(temp, "==== IP address:");
+  }
+  ACE_OS::freeaddrinfo(res);
+}
+
+void address_info() {
+  size_t addr_count;
+  ACE_INET_Addr *addr_array = 0;
```
Comment (sonndinh) [inline tests/DCPS/AddressLookup/main.cpp:107]: An ASAN build complains about leaking this array. So we probably want to add something like these from NetworkResource.cpp: https://github.com/objectcomputing/OpenDDS/blob/master/dds/DCPS/NetworkResource.cpp#L89-L101

**Review by sonndinh — CHANGES_REQUESTED** [review]

File: tests/DCPS/AddressLookup/run_test.pl [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+use lib "$ACE_ROOT/bin";
+use PerlDDS::Run_Test;
+use FileHandle;
+use Cwd;
+use strict;
+
+my $test = new PerlDDS::TestFramework();
+$test->process("AddressLookup", "main", "");
+$test->start_process("AddressLookup");
+my $retcode = $test->finish(15);
```
Comment (sonndinh) [inline tests/DCPS/AddressLookup/run_test.pl:21]: The timeout value may need to be increased. I saw the test got timeout error for test_w22_x86_i0_sec on a previous GHA run. Other than that, it looks good.

**Review by sonndinh — APPROVED** [review]

**Pull request merged**